Add unit test on export_opnames with interface.

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31531 Add unit test on export_opnames with interface.**

As suggested by @suo , add unit test on torch.jit.export_opnames with interface. A submodule is annotated as interface and assigned to an instance, and then re-assigned to another instance. Make sure the operator names are also updated.

Differential Revision: [D19539129](https://our.internmc.facebook.com/intern/diff/D19539129)

Commit:
Add unit test on export_opnames with interface.

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit b1a55ca4:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=b1a55ca44898098bb8fddd3ccb99b74aeb44f3b0).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4041337) pytorch_linux_xenial_py2_7_9_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78510204) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78510204))
<details>
<summary>
<code>Dec 20 23:48:59 RuntimeError: test_jit failed!</code>
</summary>

```
Dec 20 23:48:59 Running test_jit ... [2019-12-20 23:48:59.610064] 
Dec 20 23:48:59   File "test_jit.py", line 3771 
Dec 20 23:48:59     sub : OneTwoModule 
Dec 20 23:48:59         ^ 
Dec 20 23:48:59 SyntaxError: invalid syntax 
Dec 20 23:48:59 Traceback (most recent call last): 
Dec 20 23:48:59   File "test/run_test.py", line 456, in <module> 
Dec 20 23:48:59     main() 
Dec 20 23:48:59   File "test/run_test.py", line 449, in main 
Dec 20 23:48:59     raise RuntimeError(message) 
Dec 20 23:48:59 RuntimeError: test_jit failed! 
Dec 20 23:48:59 + cleanup 
Dec 20 23:48:59 + retcode=1 
Dec 20 23:48:59 + set +x 
Dec 20 23:48:59 =================== sccache compilation log =================== 
Dec 20 23:48:59 ERROR:sccache::server: Compilation failed: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/tmp/torch_extensions/test_compilation_error_formatting/main.cpp: In function \'int main()\':\n/tmp/torch_extensions/test_compilation_error_formatting/main.cpp:2:23: error: expected \';\' before \'}\' token\n int main() { return 0 }\n                       ^\n" } 
Dec 20 23:48:59  
Dec 20 23:48:59 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 20 23:48:59 Compile requests                 46 
Dec 20 23:48:59 Compile requests executed        27 
Dec 20 23:48:59 Cache hits                       20 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

suo commented on test/test_jit.py:
```diff
@@ -3725,6 +3725,70 @@ def forward(self, x):
+                return self.two(self.one(x, x))
+
+        class M(nn.Module):
+            sub : OneTwoModule
```
This test needs to be moved to test_jit_py3 otherwise it will fail py2 checks.

Commit:
Update on "Add unit test on export_opnames with interface."


As suggested by @suo , add unit test on torch.jit.export_opnames with interface. A submodule is annotated as interface and assigned to an instance, and then re-assigned to another instance. Make sure the operator names are also updated.

suo reviewed (approved):
thanks for adding the test!

Commit:
Update on "Add unit test on export_opnames with interface."


As suggested by @suo , add unit test on torch.jit.export_opnames with interface. A submodule is annotated as interface and assigned to an instance, and then re-assigned to another instance. Make sure the operator names are also updated.

Differential Revision: [D19539129](https://our.internmc.facebook.com/intern/diff/D19539129)
